### PR TITLE
HPCC-14243 Add method to retrieve cluster group

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -9131,6 +9158,9 @@ class CInitGroups
 
     bool loadMachineMap()
     {
+        if (machinesLoaded)
```
Comment (ghalliday) [inline dali/base/dadfs.cpp]: Does this need protecting with a critical section?

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -9131,6 +9158,9 @@ class CInitGroups
...
+            return true;
+        machinesLoaded = true;
```
Comment (ghalliday) [inline dali/base/dadfs.cpp]: Should this be delayed until the environment has been successfully read?

File: dali/base/dadfs.cpp [hunk 3]
```diff
@@ -8976,7 +8976,18 @@ class CInitGroups
     CConnectLock groupsconnlock;
     StringArray clusternames;
     unsigned defaultTimeout;
+    bool machinesLoaded;
 
+    GroupType getGroupType(const char *type)
+    {
+
```
Comment (ghalliday) [inline dali/base/dadfs.cpp]: trivial: extra blank line

File: dali/base/dadfs.cpp [hunk 4]
```diff
@@ -8976,7 +8976,18 @@ class CInitGroups
...
+        if (0 == strcmp("ThorCluster", type))
+            return grp_thor;
+        else if (0 == strcmp("RoxieCluster", type))
+            return grp_roxie;
+        else
```
Comment (ghalliday) [inline dali/base/dadfs.cpp]: Should this also support grp_hthor?

**Comment by ghalliday** [comment]
@jakesmith a few questions.  It looks a relatively safe change.

File: dali/base/dadfs.cpp (hunk 4 above)
Comment (jakesmith) [inline dali/base/dadfs.cpp]: I'm not sure, HThor 'clusters' aren't really clusters. The groups associated with them are a little odd, e.g. if you have a agent defined with multiple instances it creates a group per instance.
I'm not sure what you'd return as a group in that case, probably a group of all instances, but that wouldn't be much practical use to anyone I think.
I think I'll leave this anomaly for now, until/if anyone cares/needs it.

File: dali/base/dadfs.cpp (hunk 1 above)
Comment (jakesmith) [inline dali/base/dadfs.cpp]: don't think so, CInitGroups is a utility class used and not thread safe in general and not expected to be.

File: dali/base/dadfs.cpp (hunk 2 above)
Comment (jakesmith) [inline dali/base/dadfs.cpp]: maybe would look more logical if was at end of method.
Shouldn't make any difference as not protected/not expected to be thread-safe

**Comment by jakesmith** [comment]
@ghalliday - please see replies and minor changes.

**Pull request merged**